tests/Tests/200__monitor_and_manage/200__metrics/200__metrics.robot

```diff
@@ -1,26 +1,136 @@
+
 
 *** Variables ***
+@{RECORD_GROUPS}    Availability Metrics    SLOs - JupyterHub    SLOs - ODH Dashboard    SLOs - RHODS Operator    SLOs - Traefik Proxy    Usage Metrics
 
-@{RECORD_GROUPS}  Availability Metrics  SLOs - JupyterHub  SLOs - ODH Dashboard  SLOs - RHODS Operator  SLOs - Traefik Proxy  Usage Metrics
+@{ALERT_GROUPS}     Builds                  DeadManSnitch    RHODS-PVC-Usage    SLOs-haproxy_backend_http_responses_total    SLOs-probe_success
 
-@{ALERT_GROUPS}  Builds  DeadManSnitch  RHODS-PVC-Usage  SLOs-haproxy_backend_http_responses_total  SLOs-probe_success
 
 *** Test Cases ***
-
 Test Existence of Prometheus Alerting Rules
-  [Tags]  Smoke  Sanity  ODS-509
-  Check Prometheus Alerting Rules
+    [Tags]    Smoke
+    ...       Sanity
+    ...       ODS-509
+    Check Prometheus Alerting Rules
 
 Test Existence of Prometheus Recording Rules
-  [Tags]  Smoke  Sanity  ODS-510
-  Check Prometheus Recording Rules
+    [Tags]    Smoke
+    ...       Sanity
+    ...       ODS-510
+    Check Prometheus Recording Rules
+
+Test Metric "Notebook CPU Usage" On ODS Prometheus
+    [Documentation]    Verifing the notebook cpu usage showing on RHODS promethues
+    [Tags]    Sanity
+    ...       Tier1
+    ...       ODS-178
+    ${cpu_usage_before} =    Read Current CPU Usage
+    Run Jupyter Notebook For 5 Minutes
+    ${cpu_usage_after} =    Read Current CPU Usage
+    Should Not Be Equal    ${cpu_usage_before}    ${cpu_usage_after}
+
+Test Metric "Rhods_Total_Users" On ODS Prometheus
+    [Documentation]    Verifies that metrics rhods_total_users and jupyterhub_total_users have the same value in Prometheus
+    [Tags]    Sanity
+    ...       Tier1
+    ...       ODS-628
+    ${expression} =    Set Variable    rhods_total_users&step=1  #step = 1 beacuase it will return value of current second
+    ${rhods_total_users} =    Prometheus.Run Query    ${RHODS_PROMETHEUS_URL}    ${RHODS_PROMETHEUS_TOKEN}    ${expression}
+    ${expression} =    Set Variable    jupyterhub_total_users&step=1  #step = 1 beacuase it will return value of current second
+    ${jupyterhub_total_users} =    Prometheus.Run Query    ${RHODS_PROMETHEUS_URL}    ${RHODS_PROMETHEUS_TOKEN}
+    ...    ${expression}
+    Should Be Equal    ${rhods_total_users.json()["data"]["result"][0]["value"][-1]}
+    ...    ${jupyterhub_total_users.json()["data"]["result"][0]["value"][-1]}
+    ...    msg=metrics rhods_total_users and jupyterhub_total_users don't have the same value
+
+Test Metric "Rhods_Aggregate_Availability" On ODS Prometheus
+    [Documentation]    Verifies the rhods aggregate availability on rhods prometheus
+    [Tags]    Sanity
+    ...       Tier1
+    ...       ODS-636
+    ${expression} =    Set Variable    rhods_aggregate_availability&step=1
+    ${resp} =    Prometheus.Run Query    ${RHODS_PROMETHEUS_URL}    ${RHODS_PROMETHEUS_TOKEN}    ${expression}
+    Should Be Equal    ${resp.json()["data"]["result"][0]["value"][-1]}    1
```

Comment:
not sure about what resp variable contains, so sorry if it is a dummy question. But are you checking only the last value contained in the list "value"? If yes, wouldn't make more sense to check that the metric is always 1, then there are not periods of unavailability

Reply:
actually @bdattoma , I have set step = 1 because of that we get only one value for that instance/ second
` "value":[1648728576.457, "1"] ` here 1648728576.457 is time and "1" is value so that is why I am checking the last value of the list
so we are getting the value of availability for that second
I hope I have cleared if not feel free to ask..

Reply:
It is possible that the value will be 0 when RHODS dashboard or jupyterhub is down. We will now more about this when we run the tests within the pipeline some times

Reply:
if the test is about testing existence of the metric, then the metric exists even if its last value is 0 rather than 1, so maybe the check to apply at like 57 should be more something like `check the metric has values Not null`. But this depends on how the "existence" of a metric is defined.

Reply:
test is checking for services up and running
@bdattoma we are checking for ` Value is usually=1 (all services up and running) `  , all services up and running
https://polarion.engineering.redhat.com/polarion/#/project/OpenDataHub/workitem?id=ODS-636

Reply:
IMO, the existence of the metric should be independent from the its value. I don't see much value in checking only the last value to be honest, because the metric could be 0 or Null for all the t-1 instant, and then become 1 at the end. A better value verification would be to check if the metric has only values in a specific range (i.e., [0-1]) and never Null. 

The check on the expected value of the metric is something related to the scenarios. The expected value may change based on the RHODS service status:
- when RHODS is working fine then the metric value should be 1
- when RHODS is down, then the metric value should be 0

Saying the expected value of the metric should be 1 is assuming RHODS is up and running

@jgarciao  @rkamble5

Reply:
I'm ok with checking that the value is 1 or 0  (Polarion can be updated to reflect that).  

We can test that the value is 1 in a SLO test